packages/frontend/src/components/wallet-migration/modals/RotateKeysModal/RotateKeysModal.jsx

```diff
@@ -0,0 +1,300 @@
+                const keyType = await wallet.getAccountKeyType(accountId);
+                const accountBalance = await wallet.getBalance(keyType.accountId);
+                return { accountId, keyType, accountBalance };
+            };
+            const accountWithDetails = await Promise.all(
+                accounts.map(getAccountDetails)
+            );
+            localDispatch({
+                type: ACTIONS.ADD_ACCOUNTS,
+                accounts: accountWithDetails.reduce(((acc, { accountId, keyType, accountBalance }) => keyType == WalletClass.KEY_TYPES.FAK && accountBalance.balanceAvailable >= MINIMIM_ACCOUNT_BALANCE  ? acc.concat({ accountId, status: null }) : acc), [])
+            });
+            setLoadingEligibleRotatableAccounts(false);
+            generateAndSetPhrase();
+        };
+        setLoadingEligibleRotatableAccounts(true);
+        importRotatableAccounts();
+    }, []);
+
+    const currentAccount = useMemo(() =>  state.accounts.find((account) => account.status === IMPORT_STATUS.PENDING), [ state.accounts]);
+    const currentFailedAccount = useMemo(() => state.accounts.every((account) => account.status !== IMPORT_STATUS.PENDING) && state.accounts.find((account) => account.status === IMPORT_STATUS.FAILED),[ state.accounts]);
+
+    const batchKeyRotationNotStarted = useMemo(() => state.accounts.every((account) => account.status === null), [state.accounts]);
+    const completedWithSuccess = useMemo(() => {
+        return !loadingEligibleRotatableAccounts && (state.accounts.every((account) => account.status === IMPORT_STATUS.SUCCESS || account.status === IMPORT_STATUS.FAILED) && state.accounts[state.accounts.length - 1].status !==  IMPORT_STATUS.FAILED);
+    } , [state.accounts, loadingEligibleRotatableAccounts]);
+
+    useEffect(() => {
+        if (batchKeyRotationNotStarted) {
+            initialAccountId.current = initialAccountIdOnStart;
+        }
+    },[initialAccountIdOnStart, batchKeyRotationNotStarted]);
+
+
+    useEffect(() => {
+        if (completedWithSuccess) {
+            handleSetActiveView(WALLET_MIGRATION_VIEWS.MIGRATE_ACCOUNTS);
+        }
+    }, [completedWithSuccess]);
+
+    const handleConfirmPassphrase = async () => {
+        try {
+            const account = await wallet.getAccount(currentAccount.accountId);
+            await account.addKey(currentRecoveryKeyPair.getPublicKey());
+            await wallet.saveAccount(currentAccount.accountId, currentRecoveryKeyPair);
+            localDispatch({ type: ACTIONS.SET_CURRENT_DONE });
+            setShowConfirmSeedphraseModal(() => false);            
+        } catch (e) {
+            localDispatch({ type: ACTIONS.SET_CURRENT_FAILED_AND_END_PROCESS });
+            dispatch(showCustomAlert({
+                errorMessage: e.message,
+                success: false,
+                messageCodeHeader: 'error'
+            }));
+            await new Promise((r) => setTimeout(r, 3000));
```

Comment:
Can we please make it more clear why this `setTimeout` is needed?

Reply:
Because of rerenders, the error message flashes without the user having enough time to see the error itself. This lets the error message persist for at least 3 seconds.